### PR TITLE
Add: product-page

### PR DESCRIPTION
**Pull request by gauravsinhaweb**
Files to check:

- product
- productDetail
- ProductDisplay

### CODE REVIEW

File: src/pages/product/Product.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  getSortedProducts,
+  IncludeOutOfStockHandler,
+  priceRangeHandler,
+  getProductByCategory,
+  getSortedRating,
+} from "../../utils/utils-index";
+import "./Product.css";
+import { ProductDisplay } from "./ProductDisplay";
+import { Sidebar } from "./Sidebar";
+
```
Comment (eakansh125) [inline src/pages/product/Product.jsx:16]: Just a suggestion to import common files in a single file and then export it from that file thereby reducing the import statements in each files.

**Review by eakansh125 — APPROVED** [review]
1. The code conventions are followed
2. The folder structuring is followed
3. CSS convention is followed

Good to be merged from me

File: src/pages/product/Product.css [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  font-size: 2.4rem;
+  cursor: pointer;
+  border: none;
+  background-color: var(--white);
+}
+.emptyWishlist {
+  color: #cccccc;
+}
+.filledWishlist {
+  color: #f22426;
```
Comment (manish-GH) [inline src/pages/product/Product.css]: try not to hard code colors. Use variables instead

File: src/pages/product/Product.css [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  top: 1rem;
+  right: 1rem;
+  z-index: var(--index-999);
+  font-size: 2.4rem;
+  cursor: pointer;
+  border: none;
+  background-color: var(--white);
+}
+.emptyWishlist {
+  color: #cccccc;
```
Comment (manish-GH) [inline src/pages/product/Product.css]: here as well

File: src/pages/product/Product.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  );
+  const priceRange = priceRangeHandler(fastDelivery, filter.priceRange);
+  const offerFilter = getProductByOffer(
+    priceRange,
+    filter.offerDealOfDay,
+    filter.offerBonanza
+  );
+  const categoryFilter = getProductByCategory(offerFilter, filter.category);
+  const ratingFilter = getSortedRating(categoryFilter, filter.rating);
+  const filterProduct = ratingFilter;
```
Comment (manish-GH) [inline src/pages/product/Product.jsx:36]: Suggestion: Put this whole thing inside a function. (line 21-36)

File: src/pages/product/ProductDisplay.jsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  };
+  return (
+    <>
+      <div>
+        <div className="card_hr flex-center card_product">
+          <div className="card_hr_img flex-center">
+            <img src={image} className="card_img_product" alt={productName} />
+          </div>
+          <div
+            className={`wishlist_icon ${isHeartClicked}`}
```
Comment (manish-GH) [inline src/pages/product/ProductDisplay.jsx:62]: Nice work  using the state for the className

**Review by manish-GH — APPROVED** [review]
Try to add the app preview next time. 
The code looks clean and easily readable. Nice work.

Just have a look at the recommendations, after that you are good to merge

File: src/pages/product/Product.css (hunk 3 above)
Comment (gauravsinhaweb) [reply src/pages/product/Product.css]: Made Changes! Thanks!

File: src/pages/product/Product.css (hunk 2 above)
Comment (gauravsinhaweb) [reply src/pages/product/Product.css]: Made Changes! Thanks!

**Pull request merged**